packages/zwave-js/src/lib/commandclass/MultilevelSwitchCC.ts

```diff
@@ -581,24 +601,49 @@ export class MultilevelSwitchCCSet extends MultilevelSwitchCC {
 	) {
 		super(driver, options);
 		if (gotDeserializationOptions(options)) {
-			// TODO: Deserialize payload
-			throw new ZWaveError(
-				`${this.constructor.name}: deserialization not implemented`,
-				ZWaveErrorCodes.Deserialization_NotImplemented,
+			validatePayload(this.payload.length >= 1);
+			this.currentValue = parseMaybeNumber(
+				this.payload[0],
+				driver.options.preserveUnknownValues,
 			);
+
+			if (this.version >= 2 && this.payload.length >= 2) {
+				this.targetValue = parseNumber(this.payload[0]);
+				this.duration = Duration.parseReport(this.payload[1]);
+			}
 		} else {
 			this.targetValue = options.targetValue;
 			this.duration = options.duration;
 		}
 	}
 
-	public targetValue: number;
-	public duration: Duration | undefined;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyLevel,
+		label: "Current value",
+	})
+	public readonly currentValue: Maybe<number> | undefined;
+
+	@ccValue({ forceCreation: true })
+	@ccValueMetadata({
+		...ValueMetadata.Level,
+		label: "Target value",
+	})
+	public readonly targetValue: number | undefined;
+
+	@ccValue({ minVersion: 2 })
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyDuration,
+		label: "Remaining duration until target value",
+	})
+	public readonly duration: Duration | undefined;
```

Comment:
This seems wrong. A `MultilevelSwitchCCSet` doesn't have a `currentValue` property and its properties shouldn't be readonly.